core/shared/src/main/scala/scalaz/zio/TaskR.scala

```diff
@@ -0,0 +1,370 @@
+  final def allowInterrupt: UIO[Unit] =
+    ZIO.allowInterrupt
+
+  def apply[A](a: => A): Task[A] = effect(a)
+
+  /** See [[scalaz.zio.ZIO.access]]
+   */
+  final def access[R]: ZIO.AccessPartiallyApplied[R] =
+    ZIO.access
+
+  /** See [[scalaz.zio.ZIO.accessM]]
+   */
+  final def accessM[R]: ZIO.AccessMPartiallyApplied[R] =
+    ZIO.accessM
+
+  /** See [[scalaz.zio.ZIO.bracket]]
+   */
+  final def bracket[R, A](acquire: TaskR[R, A]): ZIO.BracketAcquire[R, Throwable, A] =
+    ZIO.bracket(acquire)
+
+  /** See [[scalaz.zio.ZIO.bracket]]
+   */
+  final def bracket[R, A, B](
+    acquire: TaskR[R, A],
+    release: A => ZIO[R, Nothing, _],
+    use: A => TaskR[R, B]
+  ): TaskR[R, B] = ZIO.bracket(acquire, release, use)
+
+  /** See [[scalaz.zio.ZIO.bracketExit]]
+   */
+  final def bracketExit[R, A](acquire: TaskR[R, A]): ZIO.BracketExitAcquire[R, Throwable, A] =
+    ZIO.bracketExit(acquire)
+
+  /** See [[scalaz.zio.ZIO.bracketExit]]
+   */
+  final def bracketExit[R, A, B](
+    acquire: TaskR[R, A],
+    release: (A, Exit[Throwable, B]) => ZIO[R, Nothing, _],
+    use: A => TaskR[R, B]
+  ): TaskR[R, B] =
+    ZIO.bracketExit(acquire, release, use)
+
+  /** See [[scalaz.zio.ZIO.checkInterruptible]]
+   */
+  final def checkInterruptible[R, A](f: Boolean => TaskR[R, A]): TaskR[R, A] =
+    ZIO.checkInterruptible(f)
+
+  /** See [[scalaz.zio.ZIO.children]]
+   */
+  final def children: UIO[IndexedSeq[Fiber[_, _]]] = ZIO.children
+
+  /** See [[scalaz.zio.ZIO.collectAll]]
+   */
+  final def collectAll[R, A](in: Iterable[TaskR[R, A]]): TaskR[R, List[A]] =
+    ZIO.collectAll(in)
+
+  /** See [[scalaz.zio.ZIO.collectAllPar]]
+   */
+  final def collectAllPar[R, A](as: Iterable[TaskR[R, A]]): TaskR[R, List[A]] =
+    ZIO.collectAllPar(as)
+
+  /** See [[scalaz.zio.ZIO.collectAllParN]]
+   */
+  final def collectAllParN[R, A](n: Long)(as: Iterable[TaskR[R, A]]): TaskR[R, List[A]] =
+    ZIO.collectAllParN(n)(as)
+
+  /** See [[scalaz.zio.ZIO.descriptor]]
+   */
+  final def descriptor: UIO[Fiber.Descriptor] = ZIO.descriptor
+
+  /** See [[scalaz.zio.ZIO.descriptorWith]]
+   */
+  final def descriptorWith[R, A](f: Fiber.Descriptor => TaskR[R, A]): TaskR[R, A] =
+    ZIO.descriptorWith(f)
+
+  /** See [[scalaz.zio.ZIO.die]]
+   */
+  final def die(t: Throwable): UIO[Nothing] = ZIO.die(t)
+
+  /** See [[scalaz.zio.ZIO.dieMessage]]
+   */
+  final def dieMessage(message: String): UIO[Nothing] = ZIO.dieMessage(message)
+
+  /** See [[scalaz.zio.ZIO.done]]
+   */
+  final def done[A](r: Exit[Throwable, A]): Task[A] = ZIO.done(r)
+
+  /** See [[scalaz.zio.ZIO.effect]]
+   */
+  final def effect[A](effect: => A): Task[A] = ZIO.effect(effect)
+
+  /** See [[scalaz.zio.ZIO.effectAsync]]
+   */
+  final def effectAsync[A](register: (Task[A] => Unit) => Unit): Task[A] =
+    ZIO.effectAsync(register)
+
+  /** See [[scalaz.zio.ZIO.effectAsyncMaybe]]
+   */
+  final def effectAsyncMaybe[A](register: (Task[A] => Unit) => Option[Task[A]]): Task[A] =
```

Comment:
I think this one can have the `R` parameter and use `TaskR[R, A]` (a change by @iravid made this possible).